lib/core/modules/coderunner/codeRunner.js

```diff
@@ -59,19 +60,23 @@ class CodeRunner {
     this.runCode.registerVar(varName, code);
   }
 
-  evalCode(code, cb, forConsoleOnly = false) {
+  async evalCode(code, cb, forConsoleOnly = false) {
     cb = cb || function() {};
     const awaitIdx = code.indexOf('await');
+    let awaiting = false;
+
     if (awaitIdx > -1) {
-      if (awaitIdx < 2) {
-        let end = code.length;
-        if (code[end - 1] === ';') {
-          end--; // Remove the `;` because we add function calls
-        }
-        code = code.substring(5, end); // remove await keyword
+      awaiting = true;
+      const instructions = Utils.compact(code.split(';'));
+      const last = instructions.pop();
+
+      if (!last.trim().startsWith('return')) {
+        instructions.push(`return ${last}`); 
       } else {
-        code = `(async function() {${code}})();`;
+        instructions.push(last); 
       }
+      
+      code = `(async function() {${instructions.join(';')}})();`;
```

Comment:
add a return to the last instruction that is not empty and wrap it into an async function